server/util/grpc_server/grpc_server.go

```diff
@@ -68,112 +68,81 @@ func MaxRecvMsgSizeBytes() int {
 	return *gRPCMaxRecvMsgSizeBytes
 }
 
-type RegisterServices func(server *grpc.Server, env *real_environment.RealEnv)
+type Builder struct {
+	env  environment.Env
+	port int
+	ssl  bool
 
-func RegisterGRPCServer(env *real_environment.RealEnv, regServices RegisterServices) error {
-	grpcServer, err := newGRPCServer(env, *gRPCPort, nil, regServices)
-	if err != nil {
-		return err
-	}
-	env.SetGRPCServer(grpcServer)
-	if *gRPCOverHTTPPortEnabled {
-		env.SetMux(&gRPCMux{
-			env.GetMux(),
-			grpcServer,
-		})
-	}
-	return nil
+	hostPort string
+	server   *grpc.Server
 }
 
-func RegisterGRPCSServer(env *real_environment.RealEnv, regServices RegisterServices) error {
-	if !env.GetSSLService().IsEnabled() {
-		return nil
-	}
-	creds, err := env.GetSSLService().GetGRPCSTLSCreds()
-	if err != nil {
-		return status.InternalErrorf("Error getting SSL creds: %s", err)
-	}
-	grpcsServer, err := newGRPCServer(env, *gRPCSPort, grpc.Creds(creds), regServices)
-	if err != nil {
-		return err
-	}
-	env.SetGRPCSServer(grpcsServer)
-	return nil
-}
-
-func RegisterInternalGRPCServer(env *real_environment.RealEnv, regServices RegisterServices) error {
-	if *internalGRPCPort == 0 {
-		return nil
-	}
-
-	grpcServer, err := newGRPCServer(env, *internalGRPCPort, nil, regServices)
-	if err != nil {
-		return err
-	}
-	env.SetInternalGRPCServer(grpcServer)
-	return nil
+func (b *Builder) GetServer() *grpc.Server {
+	return b.server
 }
 
-func RegisterInternalGRPCSServer(env *real_environment.RealEnv, regServices RegisterServices) error {
-	if *internalGRPCSPort == 0 {
-		return nil
+func NewBuilder(env environment.Env, port int, ssl bool) (*Builder, error) {
+	b := &Builder{env: env, port: port, ssl: ssl}
+	if ssl && !env.GetSSLService().IsEnabled() {
+		return nil, nil
```

Comment:
returning (nil, nil) is kind of unusual -- why is that the right thing here?

Reply:
I was going to defend this decision, but then I realized I can just check whether the SSL service is enabled before constructing this server in libmain.go, so I did that instead.